File: container/src/main/java/io/art/linux/local/container/graal/GraalLxcFeature.java
package io.art.linux.local.container.graal;

import com.oracle.svm.core.annotate.*;
import org.graalvm.nativeimage.hosted.*;
import static io.art.core.graal.GraalNativeRegistrator.*;
import static io.art.linux.local.container.graal.GraalLxcConstants.*;

@AutomaticFeature
public class GraalLxcFeature implements Feature {
    @Override
    public void beforeAnalysis(BeforeAnalysisAccess access) {
        registerStaticNonJniLibrary(access, LXC_NAME);
    }
}

File: container/src/main/kotlin/io/art/linux/local/container/graal/GraalLxcFeature.kt
package io.art.linux.local.container.graal

import com.oracle.svm.core.annotate.AutomaticFeature
import com.oracle.svm.hosted.FeatureImpl
import com.oracle.svm.hosted.c.NativeLibraries
import org.graalvm.nativeimage.hosted.Feature

@AutomaticFeature
class GraalLxcFeature : Feature {
    override fun beforeAnalysis(access: Feature.BeforeAnalysisAccess) {
        val nativeLibraries: NativeLibraries = (access as FeatureImpl.BeforeAnalysisAccessImpl).getNativeLibraries()
        nativeLibraries.addStaticNonJniLibrary("lxc")
    }
}
